### ISSUE TITLE
Smoothing bug

### ISSUE THREAD

**Issue by CelestialWalrus** [karlstav/cava#7]
http://gfycat.com/WelltodoInfatuatedGypsymoth -- note that the music is paused.

I use mpd's fifo: `cava -i fifo -p /tmp/mpd.fifo -b 60`

**Comment by livibetter** [comment]
Off-topic: I swear I didn't laugh out loud when I clicked on the GIF #cavatwitcheffect

**Comment by karlstav** [comment]
no laughing matter, or is it...

I was able to reproduce the problem. It seems that when you pause MPD it stops sending data to the fifo. So instead of getting all zeroes as one would get from alsa, it just waits on 'fread' and whatever was in the shared buffer when paused remains there. I'm trying to write m way around it by using a non-blocking file descriptor, but it's a mess. I think using select() and a timeout is the solution. I'll have to look more into it later.

**Comment by karlstav** [comment]
I was able to fix it by using a non-blocking read(), turns out it wasn't a mess at all. Closing issue.

**Comment by anko** [comment]
@karlstav Friendly tip: You can also [close issues through a commit message](https://help.github.com/articles/closing-issues-via-commit-messages/), which creates a nice link to the related commit for future reference.

**Comment by karlstav** [comment]
@anko cool, I'm still learning

**Comment by livibetter** [comment]
Were 99cb8f4ec8f23c15862d8ee80cd26a837530b37a and f0b9d40fd8112004964da2ff730f21f5769c00a2 having some hiccups? Or result of some changes in `astyle` configuration?

Either way, this would make harder to spot what exactly changed for this issue.

If I `git diff 4f08afd f0b9d40` that looks more normal. When this kind of things happen and you can try to use `git commit --amend` if you haven't pushed, even you do, I would say it's still save to do a `--force` push if you correct in a few minutes.

Frankly, if that's really the configuration, I would suggest to ditch `astyle`, I really don't believe using that would help for languages like C, our eyes, hands, and brain are the best tool.

`git diff` and check every line before commit.

**Comment by karlstav** [comment]
Ok so here is what happened:

After fixing the issue i decided to style it again with astyle. Then I realized my horrible mistake, I had styled it according to an older setting (with whitespaces as indent). So i restyled it and committed again :/ .

So thanks to @livibetter I have no learned three thing:

1) --amend lets you "regret" mistakes you made
2) ALWAYS run `git diff` before you commit
3) astyle is a "lazy man's tool" and it's much better to have control of your code "by hand".
